insights/parsers/horizon_local_settings_conf.py

```diff
@@ -0,0 +1,242 @@
+add_filter(Specs.horizon_local_settings_conf, FILTERS)
+
+
+@parser(Specs.horizon_local_settings_conf)
+class HorizonLocalSettingsConf(Parser):
+    """
+       Class to parse ``local_settings`` openstack-dashboard configuration file.
+
+       Generally, local_settings file contains data in key=value format with
+       values having data types as string, numbers, list, tuple or dict.
+       The class provides attribute ``settings`` as dictionary with lines parsed
+       line by line based on keys.
+
+       Attributes:
+           settings (dict): A dictionary containing all the configuration parameters and values.
+               Sample data::
+
+                   {
+                       'DEBUG': 'False',
+                       'LOGOUT_URL': '/dashboard/auth/logout/',
+                       'SECURE_PROXY_SSL_HEADER': ['HTTP_X_FORWARDED_PROTO', 'https'],
+                       'OPENSTACK_API_VERSIONS': {'identity': '3'},
+                       'OPENSTACK_KEYSTONE_MULTIDOMAIN_SUPPORT': 'True',
+                       'OPENSTACK_KEYSTONE_DEFAULT_DOMAIN': 'Default',
+                       'HORIZON_CONFIG["images_panel"]': 'legacy',
+                       'OPENSTACK_KEYSTONE_BACKEND': {
+                           'can_edit_domain': 'True',
+                           'can_edit_group': 'True',
+                           'can_edit_project': 'True',
+                           'can_edit_role': 'True',
+                           'can_edit_user': 'True',
+                           'name': 'native'
+                        },
+                        'API_RESULT_LIMIT': '1000',
+                        'POLICY_FILES_PATH': '/etc/openstack-dashboard',
+                        'REST_API_REQUIRED_SETTINGS': ['OPENSTACK_HYPERVISOR_FEATURES', 'LAUNCH_INSTANCE_DEFAULTS', 'OPENSTACK_IMAGE_FORMATS'],
+                        .......
+                   }
+    """
+
+    def parse_content(self, data):
+        """
+           Each line is searched for the first assignment operator to get the key
+           and then the values are extracted and converted to the corresponding data types.
+        """
+        # No data found or file is empty
+        if len(data) == 0:
+            raise ParseException("Empty file")
+
+        self.settings = {}
+        # Check if line does not starts with lowercase letters
+        local_settings = "\n".join([setting for setting in get_active_lines(data) if not re.match(r"^[a-z]+", setting)])
+
+        # Check if line contains a key
+        keys = []
+        for setting in local_settings.split("\n"):
+            key = re.findall(r"^[A-Z_\[\]\"]*\s*=", setting)
+            keys.extend(key) if len(key) > 0 else None
+
+        # No valid keys are found
+        if len(keys) == 0:
+            raise ParseException("No valid keys found.")
+
+        # Extract the keys and corresponding values from data and store in a dictionary
+        for key in range(len(keys)):
```

Comment:
How about just iterate the `keys` directly? since it's a list after all.
Or `for key, index in enumerate(keys)`

Reply:
Acknowledged. Changes done in the latest commit.